Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp

```diff
@@ -232,6 +256,73 @@ namespace ROS2
         return positions;
     }
 
+    JointsManipulationRequests::JointsVelocitiesMap JointsManipulationComponent::GetAllJointsVelocities()
+    {
+        JointsManipulationRequests::JointsVelocitiesMap velocities;
+        for (const auto& [jointName, jointInfo] : m_manipulationJoints)
+        {
+            velocities[jointName] = GetJointVelocity(jointName).GetValue();
+        }
+        return velocities;
+    }
+
+    AZ::Outcome<JointEffort, AZStd::string> JointsManipulationComponent::GetJointEffort(const AZStd::string& jointName)
+    {
+        if (!m_manipulationJoints.contains(jointName))
+        {
+            return AZ::Failure(AZStd::string::format("Joint %s does not exist", jointName.c_str()));
+        }
+
+        auto jointInfo = m_manipulationJoints.at(jointName);
+        float effort{ 0 };
+
+        if (!jointInfo.m_isArticulation)
+        {
```

Comment:
minor: you could add comments to this and the next return explaining why no effort aside from 0 can be returned in these cases.